### PR TITLE
Add GetCommits API

### PR DESCRIPTION
**Pull request by omerzi**
- [ ] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [ ] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---

### CODE REVIEW

File: vcsclient/gitlab.go [hunk 1]
```diff
@@ -340,32 +339,48 @@ func (client *GitLabClient) DeletePullRequestComment(ctx context.Context, owner,
...
-		return CommitInfo{}, err
+		return nil, err
 	}
 
 	listOptions := &gitlab.ListCommitsOptions{
 		RefName: &branch,
 		ListOptions: gitlab.ListOptions{
 			Page:    1,
-			PerPage: 1,
+			PerPage: 50,
```
Comment (sverdlov93) [inline vcsclient/gitlab.go:369]: lets make this 50 to be a common const for all providers

File: vcsclient/gitlab.go [hunk 2]
```diff
@@ -340,32 +339,48 @@ func (client *GitLabClient) DeletePullRequestComment(ctx context.Context, owner,
 
 // GetLatestCommit on GitLab
 func (client *GitLabClient) GetLatestCommit(ctx context.Context, owner, repository, branch string) (CommitInfo, error) {
+	commits, err := client.GetCommits(ctx, owner, repository, branch)
```
Comment (sverdlov93) [inline vcsclient/gitlab.go:342]: same code on all providers, can we unify?

File: vcsclient/github_test.go [hunk 3]
```diff
@@ -297,12 +297,50 @@ func TestGitHubClient_GetLatestCommit(t *testing.T) {
...
+	response, err := os.ReadFile(filepath.Join("testdata", "github", "commit_list_response.json"))
+	assert.NoError(t, err)
+
+	client, cleanUp := createServerAndClient(t, vcsutils.GitHub, false, response,
+		fmt.Sprintf("/repos/%s/%s/commits?page=1&per_page=50&sha=master", owner, repo1), createGitHubHandler)
+	defer cleanUp()
+
+	result, err := client.GetCommits(ctx, owner, repo1, "master")
+
+	require.NoError(t, err)
```
Comment (sverdlov93) [inline vcsclient/github_test.go:318]:
```suggestion
	assert.NoError(t, err)
```

File: vcsclient/gitlab_test.go [hunk 4]
```diff
@@ -287,9 +287,45 @@ func TestGitLabClient_GetLatestCommit(t *testing.T) {
...
+	assert.NoError(t, err)
+
+	client, cleanUp := createServerAndClient(t, vcsutils.GitLab, false, response,
+		fmt.Sprintf("/api/v4/projects/%s/repository/commits?page=1&per_page=50&ref_name=master",
+			url.PathEscape(owner+"/"+repo1)), createGitLabHandler)
+	defer cleanUp()
+
+	result, err := client.GetCommits(ctx, owner, repo1, "master")
+
+	require.NoError(t, err)
```
Comment (sverdlov93) [inline vcsclient/gitlab_test.go:306]:
```suggestion
	assert.NoError(t, err)
```

File: vcsclient/gitlab.go (hunk 2 above)
Comment (omerzi) [reply vcsclient/gitlab.go:342]: @sverdlov93, each call goes to a specific implementation in go (go-github, go-gitlab, etc.)

**Pull request merged**